frontend/src/lib/components/NavPanelAdvertisement/NavPanelAdvertisementLogic.ts

```diff
@@ -0,0 +1,27 @@
+import { actions, kea, key, path, props, reducers } from 'kea'
+
+import type { navPanelAdvertisementLogicType } from './NavPanelAdvertisementLogicType'
+
+export type NavPanelAdvertisementLogicProps = {
+    productKey: string
+}
+
+export const navPanelAdvertisementLogic = kea<navPanelAdvertisementLogicType>([
+    path(['lib', 'components', 'NavPanelAdvertisementLogic']),
+    props({} as NavPanelAdvertisementLogicProps),
+    key(({ productKey }) => productKey),
+    actions(() => ({
+        hideAdvertisement: () => true,
+        showAdvertisement: () => false,
+    })),
+    reducers(() => ({
+        hidden: [
+            false as boolean,
+            { persist: true },
```

Comment:
Should probably make the persistence keyed by product? I'm not certain how `kea-local-storage` works, but I'm afraid this might hide the advertisement forever, rather than for this product only? I haven't tested this myself, though, this might be working already